### PR TITLE
Change streaming terminology to parent-child in the code

### PR DESCRIPTION
**Pull request by amoss**
##### Summary

Change all uses of "master/slave" terminology to "parent-child".

##### Test Plan

No functional differences in the code.

##### Additional Information

There are many mentions of master that I have left in place:
* References to external systems (e.g. the MySQL collector and the MacOS firmware)
* The use of `webmaster`
* Anything in URLs (e.g. references to git branches)

### CODE REVIEW

File: database/rrdhost.c [hunk 1]
```diff
@@ -838,7 +838,7 @@ struct label *load_auto_labels()
             add_label_to_list(label_list, "_virt_detection", localhost->system_info->virt_detection, LABEL_SOURCE_AUTO);
 
     label_list = add_label_to_list(
-        label_list, "_is_master", (localhost->next || configured_as_master()) ? "true" : "false", LABEL_SOURCE_AUTO);
+        label_list, "_is_parent", (localhost->next || configured_as_parent()) ? "true" : "false", LABEL_SOURCE_AUTO);
```
Comment (mfundul) [inline database/rrdhost.c:841]: What's the impact of this to compatibility?

Comment (amoss) [reply database/rrdhost.c:841]: It affects google-analytics but they are converted to indices before hitting the endpoint and nobody else uses it.

Comment (mfundul) [reply database/rrdhost.c:841]: I found a doc here: https://github.com/netdata/netdata/blob/master/docs/guides/using-host-labels.md#host-labels-in-streaming
so maybe users are already using this.

Comment (thiagoftsm) [reply database/rrdhost.c:841]: I agree with @mfundul , this looks like a compatibility break? The rest of the changes is right.

Comment (ilyam8) [reply database/rrdhost.c:841]: No worries, nobody will open an issue to return masters and slaves in 2020 😉 🤣

Comment (amoss) [reply database/rrdhost.c:841]: I've discussed it with @cakrit and agreed that the chances of people using it are minor and we are changing the terminology.

File: collectors/plugins.d/pluginsd_parser.c [hunk 2]
```diff
@@ -306,7 +306,7 @@ PARSER_RC pluginsd_chart(char **words, void *user, PLUGINSD_ACTION  *plugins_act
 
     // parse the name, and make sure it does not include 'type.'
     if (unlikely(name && *name)) {
-        // when data are coming from slaves
+        // when data are streaming from child nodes
```
Comment (cosmix) [inline collectors/plugins.d/pluginsd_parser.c:309]:
```suggestion
        // when data are streamed from child nodes
```

File: daemon/main.c [hunk 3]
```diff
@@ -36,7 +36,7 @@ void netdata_cleanup_and_exit(int ret) {
         // exit cleanly
 
         // stop everything
-        info("EXIT: stopping master threads...");
+        info("EXIT: stopping main threads...");
```
Comment (cosmix) [inline daemon/main.c:39]: Typically there's one 'main' thread and this log message can be misinterpreted/confusing.

File: streaming/stream.conf [hunk 4]
```diff
@@ -21,20 +21,20 @@
     #
     # If many are given, the first available will get the metrics.
     #
-    # PROTOCOL  = tcp, udp, or unix (only tcp and unix are supported by masters)
+    # PROTOCOL  = tcp, udp, or unix (only tcp and unix are supported by parents)
```
Comment (cosmix) [inline streaming/stream.conf:24]:
```suggestion
    # PROTOCOL  = tcp, udp, or unix (only tcp and unix are supported by parent nodes)
```

**Comment by amoss** [comment]
> A few minor comments here, but there's more references to barbaric names elsewhere:
> 
>     1. streaming/stream.conf

Could you point to where - I've been through `stream.conf.in` and changed everything I could see.

**Comment by amoss** [comment]
>     3\. collectors/proc.plugin/proc_diskstats.c

I've changed references to external entities - they have to be correct, and in those cases it is encoded in the path.

**Comment by amoss** [comment]
> * web/gui/dashboard.js (cc @jacekkolasa)
> 
> * web/gui/src/dashboard.js/main.js (@jacekkolasa)

That can be a separate PR.

**Comment by amoss** [comment]
> https://github.com/netdata/netdata/blob/parent-child/streaming/stream.conf#L144

Arghhhh! That one did not show up in a search - fixing...

**Comment by mfundul** [comment]
It's hard to keep approving this, stop fixing things!

**Pull request merged**